config/src/config/network_config.rs

```diff
@@ -342,56 +342,101 @@ impl Default for RateLimitConfig {
 
 pub type TrustedPeerSet = HashMap<PeerId, TrustedPeer>;
 
+// TODO: Combine with RoleType?
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub enum TrustedPeerRole {
+    Validator = 0,
+    ValidatorFullNode,
+    FullNode,
+}
+
+impl Default for TrustedPeerRole {
+    fn default() -> Self {
+        // Default to least trusted
+        TrustedPeerRole::FullNode
+    }
+}
+
+impl From<NetworkId> for TrustedPeerRole {
+    fn from(network_id: NetworkId) -> TrustedPeerRole {
+        // FIXME: These are currently guesstimates to get it up and running, we should make VFN an enum network
+        match network_id {
+            NetworkId::Validator => TrustedPeerRole::Validator,
+            NetworkId::Public => TrustedPeerRole::ValidatorFullNode,
+            NetworkId::Private(_) => TrustedPeerRole::ValidatorFullNode,
+        }
+    }
+}
```

Comment:
I think the way I had imagined it was:

1) if it comes from the V discovery set, it is a V
2) if it comes from the VFN discovery set, it is a VFN
3) if it is undefined, it is Downstream
4) If it is defined, it is what it is

Reply:
I've axed this, it was for some weird auto generation of networks